compiler/haskell-ide-core/src/Development/IDE/State/RuleTypes.hs

```diff
@@ -36,6 +37,17 @@ import           Development.IDE.Types.SpanInfo
 -- | Kick off things
 type instance RuleResult OfInterest = ()
 
+-- ^ A map from a file A to a file B whose scenario context should be
+-- used for executing scenarios in A. We use this when running the scenarios
+-- in transitive dependencies of the files of interest so that we only need
+-- one scenario context per file of interest.
+type instance RuleResult GetScenarioRoots = Map FilePath FilePath
```

Comment:
Move out of haskell-ide-core.

Reply:
Moved to the `.Daml` version of this module.